pytensor/tensor/rewriting/linalg.py

```diff
@@ -377,3 +379,119 @@ def local_lift_through_linalg(
                 return [block_diag(*inner_matrices)]
             else:
                 raise NotImplementedError  # pragma: no cover
+
+
+def _find_diag_from_eye_mul(potential_mul_input):
```

Comment:
Is this duplicated from the other PR?

Reply:
yea. that one isnt merged yet so i had to add this function in this one too